interface/overlaps.py

```diff
@@ -0,0 +1,48 @@
+import streamlit as st
+from annotated_text import annotated_text
+
+
+def st_annotate_highlights(text, highlights):
+    highlighted_text = text
+    for highlight in highlights:
+        # TODO should this be normalized to handle i.e. lower-case matches?
+        highlighted_text = highlighted_text.replace(
+            highlight,
+            "--hl--hl>>" + highlight + "<<hl--hl--"
```

Comment:
NVM - made more sense what I stepped through the code. I just because we split on a highlight delimiter `--hl--` doesn't clarify which elements of the array are the highlights 👍